xwayland/selection/selection.c

```diff
@@ -33,15 +46,48 @@ void xwm_selection_transfer_destroy_property_reply(
 }
 
 void xwm_selection_transfer_init(struct wlr_xwm_selection_transfer *transfer) {
+	struct wlr_xwm *xwm = transfer->selection->xwm;
+
 	transfer->wl_client_fd = -1;
+
+	transfer->window = xcb_generate_id(xwm->xcb_conn);
+	xcb_create_window(xwm->xcb_conn,
```

Comment:
OK, so this is where the leak is coming from. Maybe move this out of `xwm_selection_transfer_init` for now, because it's only used by incoming transfers? That way we don't leak it for outgoing.

Reply:
I think just not calling `xwm_selection_transfer_init` is nicer here rather than shuffling the code back and forth once the outgoing path stops using a queue (soon™).

Reply:
It seems like the plan has changed, maybe we should reconsider this?